check_limits.py

```diff
@@ -1,18 +1,26 @@
+def check_range(value,min_value,max_value=None):
+  if max_value == None:
+    return value > min_value
+  else:
+    return value < min_value or value > max_value
 
-def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
+def print_result(element):
```

Comment:
Nice isolation of reporting the result 👍 
Will help in extending to other languages.